verifiable/did_doc.go

```diff
@@ -16,7 +21,43 @@ type DIDDocument struct {
 	VerificationMethod []CommonVerificationMethod `json:"verificationMethod,omitempty"`
 	AssertionMethod    []Authentication           `json:"assertionMethod,omitempty"`
 	Authentication     []Authentication           `json:"authentication,omitempty"`
-	KeyAgreement       []interface{}              `json:"keyAgreement,omitempty"`
+	KeyAgreement       []Authentication           `json:"keyAgreement,omitempty"`
```

Comment:
backward compatibility has beed broken

Reply:
@vmidyllic we have to discuss this.

Reply:
We should not break it, cast it where you need if you can